core/src/com/softwarearchitecture/game_server/buttons/GridLayout.java

```diff
@@ -0,0 +1,139 @@
+package com.softwarearchitecture.game_server.buttons;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.softwarearchitecture.GameApp;
+import com.softwarearchitecture.math.Rectangle;
+import com.softwarearchitecture.math.Vector2;
+
+public class GridLayout {
+
+    private int columns;
+    private int rows;
+
+    public GridLayout(int columns, int rows) {
+        this.columns = columns;
+        this.rows = rows;
+    }
+
+    public int getNumberOfColumns() {
+        return columns;
+    }
+
+    public int getNumberOfRows() {
+        return rows;
+    }
+
+    public Vector2 findCenter() {
+        return new Vector2(0.5f, 0.5f);
+    }
+
+    public Vector2 findCenterGrid() {
+        return new Vector2((float) columns / 2, (float) rows / 2);
+    }
+
+    public float getGridWidth() {
+        return 1f / columns * GameApp.WIDTH;
```

Comment:
I am unsure if it is good to get the width and height of the application from the entrypoint of the application. Maybe this should not import GameApp, maybe the screenManager should have this info.

Reply:
I agree. The location of height and width should be changed, but until we have functionality to decide the size of the screen, it should be where it is.